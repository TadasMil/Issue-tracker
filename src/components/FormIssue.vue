<script setup lang="ts" name="FormIssue">
import type { IValidationField } from '@/types'
import { useVModel } from '@vueuse/core'

const props = defineProps<{
  modelValue: {
    title: string
    description: string
  }
  fields: IValidationField
}>()

const emit = defineEmits<{
  (e: 'submit', event: HTMLFormElement): void
}>()

const inputValue = useVModel(props, 'modelValue', emit)
</script>

<template>
  <b-form ref="form" @submit="(event) => emit('submit', event)">
    <b-form-input
      v-model="inputValue.title"
      id="title"
      type="text"
      :state="fields.title.valid"
      placeholder="Enter your name"
    />
    <b-form-invalid-feedback id="title-feedback">
      {{ fields.title.error }}
    </b-form-invalid-feedback>

    <div class="my-3">
      <b-form-textarea
        id="textarea"
        v-model="inputValue.description"
        placeholder="Describe the issue..."
        rows="3"
        type="textarea"
        max-rows="6"
        no-resize
        :state="fields.description.valid"
      />
      <b-form-invalid-feedback id="textarea-feedback">
        {{ fields.description.error }}
      </b-form-invalid-feedback>
    </div>

    <b-button type="submit" variant="dark">Submit</b-button>
  </b-form>
</template>
