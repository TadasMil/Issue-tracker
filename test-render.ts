import type { RenderOptions } from '@testing-library/vue'
import BootstrapVue3 from 'bootstrap-vue-3'
import { render } from '@testing-library/vue'

const customRender = (ui: any, options?: RenderOptions) =>
  render(ui, {
    ...options,
    global: {
      ...options?.global,
      plugins: [BootstrapVue3, ...(options?.global?.plugins || [])],
    },
  })

export * from '@testing-library/vue'
export { customRender as render }
