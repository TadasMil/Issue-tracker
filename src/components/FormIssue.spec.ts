import { describe, it, expect } from 'vitest'
import { render } from '../../test-render'

import FormIssue from './FormIssue.vue'

describe('FormIssue', () => {
  it('should render the default form', () => {
    const { html } = render(FormIssue, {
      props: {
        modelValue: {
          title: '',
          description: '',
        },
        fields: {
          title: {
            error: '',
            valid: null,
          },
          description: {
            error: '',
            valid: null,
          },
        },
      },
    })

    expect(html()).toMatchSnapshot()
  })

  it('should render the form with errors', () => {
    const { html } = render(FormIssue, {
      props: {
        modelValue: {
          title: '',
          description: '',
        },
        fields: {
          title: {
            error: 'Title is required',
            valid: false,
          },
          description: {
            error: 'Description is required',
            valid: false,
          },
        },
      },
    })

    expect(html()).toMatchSnapshot()
  })
})
