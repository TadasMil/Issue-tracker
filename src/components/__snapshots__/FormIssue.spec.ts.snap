// Vitest Snapshot v1

exports[`FormIssue > should render the default form 1`] = `
"<form class=\\"\\"><input id=\\"title\\" class=\\"form-control\\" type=\\"text\\" placeholder=\\"Enter your name\\">
  <div class=\\"invalid-feedback\\" id=\\"title-feedback\\"></div>
  <div class=\\"my-3\\"><textarea id=\\"textarea\\" class=\\"form-control\\" placeholder=\\"Describe the issue...\\" rows=\\"3\\" style=\\"resize: none;\\" wrap=\\"soft\\" type=\\"textarea\\" max-rows=\\"6\\"></textarea>
    <div class=\\"invalid-feedback\\" id=\\"textarea-feedback\\"></div>
  </div><button class=\\"btn btn-dark btn-md\\" type=\\"submit\\">
    <!---->
    <div class=\\"btn-content\\">Submit</div>
  </button>
</form>"
`;

exports[`FormIssue > should render the form with errors 1`] = `
"<form class=\\"\\"><input id=\\"title\\" class=\\"form-control is-invalid\\" type=\\"text\\" placeholder=\\"Enter your name\\" aria-invalid=\\"true\\">
  <div class=\\"invalid-feedback\\" id=\\"title-feedback\\">Title is required</div>
  <div class=\\"my-3\\"><textarea id=\\"textarea\\" class=\\"form-control is-invalid\\" placeholder=\\"Describe the issue...\\" aria-invalid=\\"true\\" rows=\\"3\\" style=\\"resize: none;\\" wrap=\\"soft\\" type=\\"textarea\\" max-rows=\\"6\\"></textarea>
    <div class=\\"invalid-feedback\\" id=\\"textarea-feedback\\">Description is required</div>
  </div><button class=\\"btn btn-dark btn-md\\" type=\\"submit\\">
    <!---->
    <div class=\\"btn-content\\">Submit</div>
  </button>
</form>"
`;
